internal/executor/executor.go

```diff
@@ -0,0 +1,168 @@
+package executor
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// Executor interface lets us implement dummy executors for tests
```

Comment:
this comment says this is for tests but this interface is used the the `sum` package which is not a test package. which one is it, for tests or not?

Also, why is this a package at all? As far as i can tell this is only used in the `sum` package. This should probably just be move there.

Reply:
It's external to the sum package with the expectation that there could/would be other similar utilities needing this in the future.

Reply:
what the the other utilities that will need and or implement this? if we dont have other uses we should hold off creating this.

This also is a big interface and its purpose and value add are not very clear. are we planning on replacing all calls to the `os/exec` with this? 

"Do not define interfaces before they are used: without a realistic example of usage, it is too difficult to see whether an interface is even necessary, let alone what methods it ought to contain." - https://go.dev/wiki/CodeReviewComments#interfaces

Reply:
I do not have any currently pending, so your argument about not implementing it before it's necessary is valid.   

That being said, this code was already implemented elsewhere for a very similar use case so it was adopted nearly as-is.    The presumption is that we'd need the currently unused functionality as we add more vendor utilities as we've seen in ironlib.

Reply:
im not familiar with the ironlib code base. does ironlib already have the supermicro update manager code? should we just be importing from ironlib?

I'm also confused. you say "do not have any currently pending" and "as we add more vendor utilities". How can this be both?

Reply:
Ironlib has many things, but not sum.   I was referring to the Executor/FakeExecutor stuff that's used for testing/execing in ironlib.

While I do not currently have any vendor utilities in the queue, I can envision a future with supporting other vendor's utilities where this code would be reused.

Reply:
> should we just be importing from ironlib?
ironlib is mean't to execute vendor utilities in an OS running on the host.
bmclib is mean't to execute actions/calls remotely and so I think this fits in here

Reply:
If we plan to continue using this code over the longer term vs something off the shelf as suggested by @DoctorVin , I'd argue that this should be extracted into an external package that both projects could import, but we haven't finalized those plans one way or the other at this point.